services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -427,51 +427,69 @@ open class ApplicationService(
         }
     }
 
-    private fun sendApplicationToAllu(application: ApplicationEntity): Int {
+    private fun sendToAllu(entity: ApplicationEntity): Int {
+        val application = entity.toApplication()
```

Comment:
Why map the entity to the DTO/domain object? This seems to make everything downstream just a little bit more complicated.

Reply:
Moved mapping to a later stage.

Reply:
Seems like both `updateApplicationInAllu` and `createApplicationInAllu` work just as well without mapping the entity to a domain object.

The only change is that getting hankeTunnus needs to be done through hanke: `application.hanke.hankeTunnus!!`. This could be simplified by adding a getter to ApplicationEntity for getting the hankeTunnus:
```kotlin
    fun hankeTunnus(): String = hanke.hankeTunnus!!
```
Sadly, Kotlin's lazy values don't seem to be working with JPA entities.

Reply:
Changed as suggested.